include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp

```diff
@@ -173,10 +217,52 @@ class Profile
         const trajectory::Orbit& anOrbit, const trajectory::Orbit::FrameType& anOrbitalFrameType
     );
 
+    /// @brief Construct a flight profile with custom target pointing
+    ///
+    /// @param anOrbit An orbit
+    /// @param anOrientationGenerator
+    /// @return Flight profile
+    static Profile GenerateTrackingProfile(
+        const trajectory::Orbit& anOrbit, const std::function<Quaternion(const State&)>& anOrientationGenerator
+    );
+
+    /// @brief Generate a custom orientation
+    ///
+    /// @param anAlignmentAxis An alignment axis
+    /// @param aClockingAxis A clocking axis
+    /// @param anAngularOffset An angular offset
+    /// @param aTrajectory A trajectory
+
+    static std::function<Quaternion(const State&)> GenerateCustomOrientation(
+        const Target& anAlignmentTarget,
+        const Target& aClockingTarget,
+        const Angle& anAngularOffset = Angle::Zero(),
+        const Trajectory& aTrajectory = Trajectory::Undefined()
+    );
+
    private:
     Unique<Model> modelUPtr_;
 
     Profile();
+
+    static Vector3d GetGeocentricNadirDirectionVector(const State& state);
+
+    static Vector3d GetGeodeticNadirDirectionVector(const State& state);
+
+    static Vector3d GetCelestialDirectionVector(const State& state, const Celestial& aCelestial);
+
+    static Vector3d GetVelocityDirectionVector(const State& state);
+
+    static Vector3d GetTargetDirectionVector(const State& state, const Trajectory& targetTrajectory);
+
+    static Vector3d GetClockingAxisVector(const Vector3d& alignmentAxisVector, const Vector3d& clockingVector);
+
+    static RotationMatrix GetRotationMatrix(
+        const Axis& alignmentAxis,
+        const Axis& clockingAxis,
+        const Vector3d& alignmentAxisVector,
+        const Vector3d& clockingAxisVector
+    );
```

Comment:
Shouldn't we name the input `aState`, `anAlignmentAxis`, etc to follow OSTk's conventions?